src/sinks/elasticsearch/retry.rs

```diff
@@ -14,6 +14,37 @@ struct EsResultResponse {
     items: Vec<EsResultItem>,
 }
 
+impl EsResultResponse {
+    fn parse(body: &str) -> Result<Self, String> {
+        serde_json::from_str::<EsResultResponse>(body).map_err(|json_error| {
+            format!(
+                "some messages failed, could not parse response, error: {}",
+                json_error
+            )
+        })
+    }
+
+    fn iter_status(&self) -> impl Iterator<Item = (StatusCode, Option<&EsErrorDetails>)> {
+        self.items.iter().filter_map(|item| {
+            item.result()
+                .status
+                .and_then(|status| StatusCode::from_u16(status).ok())
+                .map(|status| (status, item.result().error.as_ref()))
+        })
+    }
+
+    fn get_error_reason(&self, body: &str) -> String {
+        match self
+            .items
+            .iter()
+            .find_map(|item| item.result().error.as_ref())
```

Comment:
It is, at least if we are selecting just one error and not logging all of them which are usually just duplicates and there is usually a bunch of them, one for each failed item, which would be quite verbose.

In the case of selecting one error, when partial retry is enabled and we don't retry it is because there is no retryable error in the response so all of the errors are equally responsible/interesting so we can log any.

And when partial retry isn't enabled we don't retry on any error so the same holds.

Reply:
I added a quick comment to the function to that effect.